pandas/_libs/src/ujson/lib/ultrajsondec.c

```diff
@@ -160,12 +163,39 @@ FASTCALL_ATTR JSOBJ FASTCALL_MSVC decode_numeric(struct DecoderState *ds) {
                 // FIXME: Check for arithmetic overflow here
                 // PERF: Don't do 64-bit arithmetic here unless we know we have
                 // to
-                intValue = intValue * 10ULL + (JSLONG)(chr - 48);
-
-                if (intValue > overflowLimit) {
-                    return SetError(ds, -1, overflowLimit == LLONG_MAX
-                                                ? "Value is too big"
-                                                : "Value is too small");
+                newDigit = (JSLONG)(chr - 48);
+
+                // TO DO: need to fix overflow catching
+                if (intValue*10ULL + newDigit > overflowLimit) {
+                    // TO DO: store current intValue for later processing
+
+                    // convert current inValue into string
+                    int length = snprintf( NULL, 0, "%lu", intValue);
+                    char* intValue_asStr = malloc( length + 1 );
+                    snprintf(intValue_asStr, length + 1, "%lu", intValue);
+                    
+                    // copy current ds->cStr into a temporary variable
+                    char* cStr_existing = malloc(strlen(ds->cStr)+1);
+                    memcpy(cStr_existing, ds->cStr, strlen(ds->cStr)+1);
+                    
+                    // size of ds->cStr after concatenation with str
+                    size_t new_size = strlen(cStr_existing)+strlen(intValue_asStr)+1;
+                    
+                    char* new_cStr = malloc(new_size);
+                    memcpy(new_cStr, cStr_existing, strlen(cStr_existing));
+                    strcat(new_cStr, intValue_asStr);
+
+                    // copy concatenated string back to ds->cStr
+                    // TO DO: this is failing
+                    ds->cStr = realloc(ds->cStr, new_size);
```

Comment:
@WillAyd I know this isn't right (I get segfaults) but not sure how to fix it...